RecordCollection/Controllers/AlbumsController.cs

```diff
@@ -15,8 +19,39 @@ public AlbumsController(RecordCollectionContext context)
         public IActionResult Index()
         {
             var albums = _context.Albums.ToList();
-            
             return View(albums);
         }
+
+        [Route("Albums/{id:int}")]
+        public async Task<IActionResult> Show(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var retrieved = await _context.Albums.FindAsync(id);
+
+            var relatedAlbums = _context.Albums
+                .Where(a => a.Artist == retrieved.Artist && a.Id != retrieved.Id)
+                .OrderBy(a => a.ReleaseDate);
+
```

Comment:
We could use the extract method refactoring pattern on lines 35-37 and create a method that gets a list of related ablums to better adhere to SRP and make this action easier to read at a glance